core/metrics.go

```diff
@@ -0,0 +1,85 @@
+package core
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/navidrome/navidrome/db"
+	"github.com/navidrome/navidrome/log"
+	"github.com/navidrome/navidrome/model"
+	"github.com/navidrome/navidrome/persistence"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// typical Prometheus scrape interval is in 10..30 seconds range
+const METRICS_INTERVAL = 15 * time.Second
+
+type PrometheusMetrics struct {
+	DbTotal *prometheus.GaugeVec
+}
+
+func NewPrometheusMetrics() *PrometheusMetrics {
+	res := &PrometheusMetrics{
+		DbTotal: prometheus.NewGaugeVec(
+			prometheus.GaugeOpts{
+				Name: "db_model_totals",
+				Help: "Total number of DB items per model",
+			},
+			[]string{"model"},
+		),
+	}
+	err := prometheus.DefaultRegisterer.Register(res.DbTotal)
+	if err != nil {
+		log.Error("Unable to register db_model_totals metrics")
+		return nil
+	}
+	return res
+}
+
+func processSqlAggregateMetrics(ctx context.Context, dataStore model.DataStore, targetGauge *prometheus.GaugeVec) {
```

Comment:
I don't think it is worth to query this info every 15 seconds, as it only changes after a scan. We should update it at the end of the scan process.

What may be valuable to measure is number of scrobbled tracks per user. Even in this case, we can update the gauge when we actually receive the scrobble, instead of keep polling it. Remember SQLite3 rettention is very bad and we don't want to keep the DB busy

Reply:
I rewrite completely this PR. Worker and "every 15 seconds" are removed.

>may be valuable to measure is number of scrobbled tracks per user

I'm not sure that this is a good fit for Prometheus metrics. User's data should not affect labels space (if it's something like `{username="alex", status="scrobbled"}`). But I can counter with total scrobbled count. 

>SQLite3 rettention is very bad and we don't want to keep the DB busy

From my experience, SQLite3 is very good at concurrent "connections" if only few of them perform write operations.